lib/IRGen/CallEmission.h

```diff
@@ -56,6 +56,8 @@ class CallEmission {
   /// that it's true --- but is otherwise derivable from
   /// RemainingArgsForCallee, at least between calls.
   bool EmittedCall;
+  
+  bool IsMustTail = false;
```

Comment:
Can you add a comment here explaining what must tail does? I would also maybe put in the restrictions around it.